lib/conjur/extension/repository.rb

```diff
@@ -0,0 +1,183 @@
+# frozen_string_literal: true
+
+require_relative './extension'
+require_relative './implementation'
+
+module Conjur
+  module Extension
+    # Extension::Repository loads Conjur extensions from the file systems
+    # and allows Conjur to retrieve an interface (Extension Set) for
+    # calling methods on extension classes safely.
+    #
+    # :reek:TooManyInstanceVariables, :reek:InstanceVariableAssumption
+    class Repository
+      attr_reader :extensions
+
+      # ExtensionClass is uses to store the registered extension classes
+      # in a more strongly typed manner
+      RegisteredExtensionClass = Struct.new(
+        :kind,
+        :extension_class,
+        keyword_init: true
+      )
+
+      @loaded_extensions = []
+      @registered_extension_classes = []
+
+      class << self
+        # This are intentionally writable to support testing
+        # :reek:Attribute
+        attr_accessor :loaded_extensions, :registered_extension_classes
+
+        # :reek:LongParameterList is due to dependency injection
+        def register_extension(
```

Comment:
Since this is the Extension Repository, would it make sense to drop the `extension` prefixes and suffixes from methods and parameter names in this class? Not a huge deal

Reply:
Good call! I left `load_extension` since `#load` is an `Object` built-in (and used in that method). Attempting to rename it causes all kinds of issues... 😆